stan/math/rev/mat/meta/operands_and_partials.hpp

```diff
@@ -31,7 +32,7 @@ class ops_partials_edge<double, std::vector<var> > {
 
   void dump_partials(double* partials) {
     for (int i = 0; i < this->partials_.size(); ++i) {
-      partials[i] = this->partials_[i];
+      partials[i] = this->partials_(i);
```

Comment:
partials_ is an Eigen structure which should be traversed with the () operators.

Reply:
Maybe don't use `[]` on the L.H.S and `()` on the R.H.S ^^

I think `[]` is fine for Eigen structures. Since we know `i` will not be out of bound, we can use the `coeff` method, which doesn't check whether `i` is in bound and is therefore slightly faster.